ooniprobe/View/Informed consent/Onboarding3ViewController.m

```diff
@@ -0,0 +1,164 @@
+    [things_to_know_1 addAttribute:NSFontAttributeName
+                                value:[UIFont fontWithName:@"FiraSans-SemiBold" size:17]
+                                range:NSMakeRange(0, things_to_know_1.length)];
+    
+    [self.textLabel setAttributedText:things_to_know_1];
+    [self.nextButton setTitle:[NSLocalizedString(@"i_understand", nil) uppercaseString] forState:UIControlStateNormal];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+-(IBAction)showQuiz:(id)sender{
+    UIViewController *aViewController = [[UIViewController alloc] initWithNibName:@"Quiz" bundle:nil];
+    quizView = aViewController.view;
+    UIButton *trueButton = (UIButton*)[quizView viewWithTag:1];
+    [trueButton addTarget:self action:@selector(answer:) forControlEvents:UIControlEventTouchUpInside];
+    [trueButton setTitle:[NSLocalizedString(@"_true", nil) uppercaseString] forState:UIControlStateNormal];
+
+    UIButton *falseButton = (UIButton*)[quizView viewWithTag:2];
+    [falseButton addTarget:self action:@selector(answer:) forControlEvents:UIControlEventTouchUpInside];
+    [falseButton setTitle:[NSLocalizedString(@"_false", nil) uppercaseString] forState:UIControlStateNormal];
+
+    UIButton *closeView = (UIButton*)[quizView viewWithTag:4];
+    [closeView addTarget:self action:@selector(dismissPopup) forControlEvents:UIControlEventTouchUpInside];
+
+    UIView *cointainerWindow = (UIView*)[quizView viewWithTag:3];
+    cointainerWindow.layer.cornerRadius = 12;
+    cointainerWindow.layer.masksToBounds = true;
+    
+    UILabel *title = (UILabel*)[quizView viewWithTag:5];
+    [title setText:NSLocalizedString(@"pop_quiz", nil)];
+    [self reloadQuestion];
+    
+    PopupView* popup = [PopupView popupViewWithContentView:quizView];
+    [popup setDidFinishDismissingCompletion:^{}];
+    [popup show];
+}
+
+- (void)reloadQuestion{
+    NSMutableAttributedString *question_intro = [[NSMutableAttributedString alloc] initWithString:[NSString stringWithFormat:@"%@ %ld/2\n\n", NSLocalizedString(@"question", nil), question_number]];
+    [question_intro addAttribute:NSFontAttributeName
+                                value:[UIFont fontWithName:@"FiraSans-Regular" size:17]
+                                range:NSMakeRange(0, question_intro.length)];
+    NSMutableAttributedString *question_text;
+    if (question_number == 1)
+        question_text = [[NSMutableAttributedString alloc] initWithString:NSLocalizedString(@"question_1", nil)];
+    else if (question_number == 2)
+        question_text = [[NSMutableAttributedString alloc] initWithString:NSLocalizedString(@"question_2", nil)];
```

Comment:
Maybe you can panic or emit error if the question number is out of range?

Reply:
how can the number be out of range?

Reply:
Bad future refactoring? I guess you can use `assert` for that.